dbms/src/Flash/Coprocessor/FunctionBuildHelper.cpp

```diff
@@ -258,17 +265,24 @@ String DAGExpressionAnalyzerHelper::buildCastFunctionInternal(
 }
 
 /// buildCastFunction build tidb_cast function
-String DAGExpressionAnalyzerHelper::buildCastFunction(
+String FunctionBuildHelper::buildCastFunction(
+    const FunctionExpr & function_expr,
     DAGExpressionAnalyzer * analyzer,
-    const tipb::Expr & expr,
     const ExpressionActionsPtr & actions)
 {
+    const auto * tipb_function_expr = dynamic_cast<const TiPBExprFunctionExpr *>(&function_expr);
+    if (tipb_function_expr == nullptr)
+    {
+        // todo support non tipb function expr
```

Comment:
What happens for non tipb function expr.
Exception or incorrect result?

Reply:
For non tipb function expr, it will use `buildDefaultFunction` to build the function, for `cast` function, it means you should provide two columns: `col1` is the col to be casted, `col2` is a constant column and its value is the target type's name.

Reply:
What do you think about throwing a mistake here?
Because it may cause the results of gtest to be inconsistent with the actual results

Reply:
But we need find a way to test `cast` and `in` in unit test, some suggestions?

Reply:
Maybe I can construct a tipb expr based on the input ColumnFunctionExpr for `in` and `cast`.

Reply:
> Maybe I can construct a tipb expr based on the input ColumnFunctionExpr for `in` and `cast`.

how about constructing a tipb expr instead of using ColumnFunctionExpr in gtest for all functoins...

Reply:
Let's have a try